### PR TITLE
fix: fix the bug in style version comparison

### PR DESCRIPTION
**Pull request by WeizhengSap**
ticket: CXSPA-1467

### CODE REVIEW

File: projects/storefrontstyles/scss/_versioning.scss [hunk 1]
```diff
@@ -64,6 +64,81 @@ $useLatestStyles: false !default;
...
+  $v2Major: map-get($v2, major);
+  $v2Minor: map-get($v2, minor);
+
+  @if ($v1Major > $v2Major) {
+    @return 1;
+  }
+  @if ($v1Major < $v2Major) {
+    @return -1;
+  }
+  @if ($v1Major == $v2Major) {
```
Comment (plabadie) [inline projects/storefrontstyles/scss/_versioning.scss]: Perhaps this comparision can be removed.  If none is greater of lower, they must be equal

File: projects/storefrontstyles/scss/_versioning.scss [hunk 2]
```diff
@@ -64,6 +64,81 @@ $useLatestStyles: false !default;
...
+    @if ($v1Minor > $v2Minor) {
+      @return 1;
+    }
+    @if ($v1Minor < $v2Minor) {
+      @return -1;
+    }
+    @if ($v1Minor == $v2Minor) {
```
Comment (plabadie) [inline projects/storefrontstyles/scss/_versioning.scss]: Perhaps this comparision can be removed too . If none is greater of lower, they must be equal

File: projects/storefrontstyles/scss/_versioning.scss [hunk 3]
```diff
@@ -64,6 +64,81 @@ $useLatestStyles: false !default;
...
+
+@function toNumber($string) {
+  $strings: '0' '1' '2' '3' '4' '5' '6' '7' '8' '9';
+  $numbers: 0 1 2 3 4 5 6 7 8 9;
+
+  $result: 0;
+  @for $i from 1 through str-length($string) {
+    $character: str-slice($string, $i, $i);
+    $index: index($strings, $character);
+    $number: nth($numbers, $index);
```
Comment (plabadie) [inline projects/storefrontstyles/scss/_versioning.scss]: I think the index is already the number we want.  Maybe there is no need for `$numbers: 0 1 2 3 4 5 6 7 8 9;`

File: projects/storefrontstyles/scss/_versioning.scss [hunk 4]
```diff
@@ -64,6 +64,81 @@ $useLatestStyles: false !default;
...
+
+  @return (major: $major, minor: $minor);
+}
+
+@function toString($value) {
+  @if type-of($value) == 'number' {
+    @return #{$value};
+  } @else if type-of($value) == 'string' {
+    @return $value;
+  } @else if type-of($value) == 'list' {
```
Comment (plabadie) [inline projects/storefrontstyles/scss/_versioning.scss:103]: In which case would there be a `list`?

File: projects/storefrontstyles/scss/_versioning.scss [hunk 5]
```diff
@@ -54,7 +54,7 @@ $useLatestStyles: false !default;
 
 @mixin forVersion($from, $to: 0) {
   // We never create content that is part of future releases, this should not really happen.
-  @if ($from <= $_fullVersion) {
+  @if (compareVersion($from, $_fullVersion) <= 0) {
```
Comment (plabadie) [inline projects/storefrontstyles/scss/_versioning.scss:64]: Overall, what is the expected behaviour for patch versions?

File: projects/storefrontstyles/scss/_versioning.scss (hunk 3 above)
Comment (WeizhengSap) [reply projects/storefrontstyles/scss/_versioning.scss]: index() starts from 1, not from 0. So, e.g. for character `5`, `index$` is 6. `nth` gets the 6th number from `$numbers`, so 5 is the result. Yes, we can get `$number` from `$index - 1`.

File: projects/storefrontstyles/scss/_versioning.scss (hunk 4 above)
Comment (WeizhengSap) [reply projects/storefrontstyles/scss/_versioning.scss:103]: `1.2.3` will be `[1.2, 0.3]`; `1.2.3.4` will be `[1.2, 0.3, 0.4]`. So, customer can set any dot numbers, but we only get the first item in list `toString(nth($value, 1));`, that is `1.2`.

File: projects/storefrontstyles/scss/_versioning.scss (hunk 5 above)
Comment (WeizhengSap) [reply projects/storefrontstyles/scss/_versioning.scss:64]: patch version is ignored. `1.2.3` will be treated as `1.2`. I can add the logic to handle patch version (not hard), the thing is, do we want to do it? From my understanding, patch is only for bug fix. There should be no breaking changes involved.

**Review by plabadie — DISMISSED** [review]

**Review by plabadie — APPROVED** [review]

**Pull request merged**